docs/sources/enterprise/aws-secrets-plugin.md

```diff
@@ -0,0 +1,87 @@
+| `secret_description`    | Description applied to every secret in AWS Secrets Manager. For bookkeeping purposes only                                                                          | Metrics team datasource                  | _none_                          |
+
+## Migration
+
+You can configure Grafana to migrate your existing secrets from Grafana to the plugin on startup. This migration is a one-time blocking operation that runs on Grafana startup, meaning Grafana will not be usable until migration is complete.
+
+**Note:** The speed of this migration is contingent on the number of secrets and system performance, so it can potentially run slowly (e.g. two secrets per second).
+
+Once migration to the plugin has completed, and if [backwards compatibility](#backwards-compatibility) is disabled, the plugin will be required for Grafana to start. If the plugin is uninstalled unexpectedly, or if it fails to start for any reason, Grafana will also fail to start. Because the ability to migrate from the plugin back to Grafana is not yet implemented, we highly recommend keeping backwards compatibility turned on (default).
+
+### Migration to plugin
+
+Migration from Grafana to the plugin is initiated by configuration. Add the following to the Grafana `custom.ini` file:
+
+```ini
+[secrets]
+migrate_to_plugin = true
+```
+
+Secrets will be migrated to the plugin one-by-one, then deleted from the Grafana database if backwards compatibility is disabled.
+
+## Backwards compatibility
+
+By default, after the plugin is installed and enabled, secrets will be stored in AWS Secrets Manager **and** in a legacy location in the Grafana database. If you wish to stop to storing secrets in the legacy table while the secrets plugin is enabled, you can disable backwards compatibility by adding a [feature toggle]({{< relref "../setup-grafana/configure-grafana/#feature_toggles" >}}) called `disableSecretsCompatibility`:
+
+```ini
+[feature_toggles]
+enabled = disableSecretsCompatibility
+```
+
+While secrets will not be read from this table while the plugin is active, it gives the ability to quickly uninstall the secrets plugin without losing access to all of your secrets. Because the ability to migrate from the plugin back to Grafana is not yet implemented, we highly recommend keeping backwards compatibility turned on.
+
+## Limitations
+
+If secret rotation has been activated on your AWS Secrets Manager instance, the backwards compatibility feature will not be sufficient as a fallback for plugin errors. This is because legacy secrets will not be kept up to date with AWS Secrets Manager automatically. If you are planning to implement secret rotation, we recommend waiting until migration from the plugin back to Grafana is available if there is a desire to potentially uninstall the AWS Secrets Manager plugin.
```

Comment:
```suggestion
If your AWS Secrets Manager instance uses secret rotation, the backward compatibility feature will not be a sufficient fallback for plugin errors, because legacy secrets will not be kept up to date with AWS Secrets Manager automatically. If you plan to implement secret rotation, we recommend that you avoid installing this plugin until we implement migration from the plugin back to Grafana's database.
```